stats/registry.go

```diff
@@ -0,0 +1,51 @@
+package stats
+
+import (
+	"fmt"
+	"sync"
+)
+
+var errFmtMetricExists = "fatal: metric %q already exists"
+
+// Registry tracks metrics and reporters
+type Registry struct {
+	sync.Mutex
+	// here we use just the metric name as key. it does not include any prefix
+	// this means that technically we're more strict here then needed around naming conflicts:
+	// it's possible to have metrics with the same name, but a different prefix.
+	// we still complain about that
+	metrics map[string]GraphiteMetric
+}
+
+func NewRegistry() *Registry {
+	return &Registry{
+		metrics: make(map[string]GraphiteMetric),
+	}
+}
+
+func (r *Registry) add(name string, getMetric func() GraphiteMetric) GraphiteMetric {
```

Comment:
You're right. the idea was to save some work: if the metric already exists, we can bail out early without doing the work of instantiating the metric, and to make sure we don't accidentally create two versions of the same metric.  but since we panic anyway, and this is such a rare case, I will change it to do it like you say.